Tella/Components/SearchBarView.swift

```diff
@@ -0,0 +1,98 @@
+//
+//  SearchBarView.swift
+//  Tella
+//
+//  Created by gus valbuena on 4/10/24.
+//  Copyright © 2024 HORIZONTAL. All rights reserved.
+//
+
+
+import SwiftUI
+
+struct SearchBarView: View {
+    @Binding var searchText: String
+    var body: some View {
+        if #available(iOS 15.0, *) {
+            SearchBar(searchText: $searchText)
+        } else {
+            SearchBarNoFocus(searchText: $searchText)
+        }
+    }
+}
+
+@available(iOS 15.0, *)
+struct SearchBar: View {
+    @Binding var searchText: String
+    @FocusState private var isInputActive:Bool
+    var body: some View {
+        VStack {
+            ZStack {
+                PlaceholderText(searchText: $searchText)
+                HStack {
+                    Image("file.search")
+                    TextField("", text: $searchText)
```

Comment:
I think it's better to use `SearchTextField` and `SearchTextFieldFocused` instead of the `SearchBar` and  `SearchBarNoFocus`

```
@available(iOS 15.0, *)
struct SearchTextField: View {
    @Binding var searchText: String
    @FocusState private var isInputActive:Bool
    var body: some View {
        TextField("", text: $searchText)
            .keyboardType(.default)
            .textFieldStyle(TextfieldStyle(shouldShowError: false))
            .frame(height: 22)
            .focused($isInputActive)
    }
}

struct SearchTextFieldNoFocus: View {
    @Binding var searchText: String
    var body: some View {
        TextField("", text: $searchText)
            .keyboardType(.default)
            .textFieldStyle(TextfieldStyle(shouldShowError: false))
            .frame( height: 22)
    }
}
```

Reply:
Yeah i think about that before implementing it but then I realized I need to use isInputActive for the overlay style that is used in the HStack wrapper.

What I did is to Isolate the whole HStack and create two new components FocusedSearchBar and UnfocusedSearchbar. Which the are integrated into the SearchBarView

```
struct SearchBarView: View {
    @Binding var searchText: String
    var body: some View {
        VStack {
            ZStack {
                PlaceholderText(searchText: $searchText)
                if #available(iOS 15.0, *) {
                    FocusedSearchBar(searchText: $searchText)
                } else {
                    UnfocusedSearchBar(searchText: $searchText)
                }
            }
        }.padding(.top, 8)
    }
}
```